src/services/navigationBar.ts

```diff
@@ -534,4 +541,200 @@ namespace ts.NavigationBar {
+                case SyntaxKind.VariableDeclaration:
+                    // Only add to the navbar if at the top-level of the file
+                    // Note: "const" and "let" are also SyntaxKind.VariableDeclarations
+                    if(node.parent/*VariableDeclarationList*/.parent/*VariableStatement*/
+                           .parent/*SourceFile*/.kind !== SyntaxKind.SourceFile) {
+                        return undefined;
+                    }
+                    // If it is initialized with a function expression, handle it when we reach the function expression node
+                    const varDecl = node as VariableDeclaration;
+                    if (varDecl.initializer && (varDecl.initializer.kind === SyntaxKind.FunctionExpression ||
+                                                varDecl.initializer.kind === SyntaxKind.ArrowFunction ||
+                                                varDecl.initializer.kind === SyntaxKind.ClassExpression)) {
+                        return undefined;
+                    }
+                    // Fall through
+                case SyntaxKind.FunctionDeclaration:
+                case SyntaxKind.ClassDeclaration:
+                case SyntaxKind.Constructor:
+                case SyntaxKind.GetAccessor:
+                case SyntaxKind.SetAccessor:
+                    // "export default function().." looks just like a regular function/class declaration, except with the 'default' flag set
+                    const name = node.flags && (node.flags & NodeFlags.Default) ? "default" :
+                            node.kind === SyntaxKind.Constructor ? "constructor" :
+                            declarationNameToString((node as (Declaration)).name);
+
+                    const elementKind =
+                            node.kind === SyntaxKind.VariableDeclaration ? ScriptElementKind.variableElement :
+                            node.kind === SyntaxKind.FunctionDeclaration ? ScriptElementKind.functionElement :
+                            node.kind === SyntaxKind.ClassDeclaration ? ScriptElementKind.classElement :
+                            node.kind === SyntaxKind.Constructor ? ScriptElementKind.constructorImplementationElement :
+                            node.kind === SyntaxKind.GetAccessor ? ScriptElementKind.memberGetAccessorElement :
+                            node.kind === SyntaxKind.SetAccessor ? ScriptElementKind.memberSetAccessorElement :
+                            "unknown";
+
+                    return getNavBarItem(name, elementKind, [getNodeSpan(node)]);
+                case SyntaxKind.FunctionExpression:
+                case SyntaxKind.ArrowFunction:
+                case SyntaxKind.ClassExpression:
+                    return getDefineModuleItem(node) || getFunctionOrClassExpressionItem(node);
+                case SyntaxKind.MethodDeclaration:
+                    const methodDecl = node as MethodDeclaration;
+                    if (!methodDecl.name) {

```

Comment:
This doesn't happen (it's at worst `MissingNode` which you might need to handle in `declarationNameToString` if it doesn't already)